modules/basics/src/main/java/com/opengamma/strata/basics/schedule/PeriodicSchedule.java

```diff
@@ -424,6 +425,17 @@ public Schedule createSchedule(ReferenceData refData) {
     List<LocalDate> adj = applyBusinessDayAdjustment(unadj, refData);
     List<SchedulePeriod> periods = new ArrayList<>();
     try {
+      // Remove any duplicate adjusted dates if requested
+      if (combinePeriodsIfNecessary) {
```

Comment:
Looking at how the code turned out, it seems my analysis was wrong, sorry. I don't think it is right to add `combinePeriodsIfNecessary` as an extra property in `PeriodicSchedule` since it can be handled just as well as an extra parameter to `createSchedule()` (with a backwards compatibility overload).

This has the side effect that `PeriodicSchedule` can stay as a normal bean, removing another incompatibility.

Reply:
No sweat, does work out cleaner with the overloaded method. Though wasn't a significant saving in terms of number of new lines, so I've tried to making savings elsewhere sits at 1999 lines :flushed: